Habitica/src/main/java/com/habitrpg/android/habitica/ui/fragments/tasks/TaskRecyclerViewFragment.kt

```diff
@@ -163,12 +163,15 @@ open class TaskRecyclerViewFragment : BaseFragment<FragmentRefreshRecyclerviewBi
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         context?.let { binding?.recyclerView?.setBackgroundColor(ContextCompat.getColor(it, R.color.content_background)) }
-        if (Task.TYPE_DAILY == classType) {
-            if (user?.isValid == true && user?.preferences?.dailyDueDefaultView == true) {
-                taskFilterHelper.setActiveFilter(Task.TYPE_DAILY, Task.FILTER_ACTIVE)
+        savedInstanceState?.let { this.classType = savedInstanceState.getString(CLASS_TYPE_KEY, "") }
+
+        when (classType) {
+            Task.TYPE_TODO -> taskFilterHelper.setActiveFilter(Task.TYPE_TODO, Task.FILTER_ACTIVE)
+            Task.TYPE_DAILY -> {
+                if (user?.isValid == true && user?.preferences?.dailyDueDefaultView == true) {
+                    taskFilterHelper.setActiveFilter(Task.TYPE_DAILY, Task.FILTER_ACTIVE)
+                }

```

Comment:
We check the `classType` to decide which filters to apply. The issue was `classType` was being assigned *after* the filters are applied, so from a cold boot we'll be checking against a `null` `classType`. This hasn't been noticed before because *most* of the time a cold boot will load into the habits tab, whose default filter is "all."

Reply:
Also Kotlin-ized some things for readability